static/js/publisher/form/banner.js

```diff
@@ -0,0 +1,219 @@
+        <div className="p-notification--negative">
+          <p className="p-notification__response">
+            {Object.keys(errors).map(fileName => (
+              <Fragment key={`errors-${fileName}`}>
+                {fileName}
+                &nbsp;
+                {errors[fileName].map((error, index) => (
+                  <Fragment key={`errors-${fileName}-${index}`}>
+                    {error}
+                    <br />
+                  </Fragment>
+                ))}
+              </Fragment>
+            ))}
+          </p>
+        </div>
+      );
+    }
+    return false;
+  }
+
+  renderBackground() {
+    const { bannerImage } = this.state;
+    const { restrictions } = this.props;
+    const background = bannerImage.url;
+
+    let style;
+    let backgroundClasses = ["p-market-banner__image", "u-vertically-center"];
+    if (background) {
+      style = {
+        backgroundImage: `url(${background})`
+      };
+    } else {
+      backgroundClasses.push("is-empty");
+    }
+
+    return (
+      <Fragment>
+        <FileInput
+          inputName="banner-image"
+          active={true}
+          fileChangedCallback={this.bannerImageChangeHandler}
+          noFocus={true}
+          restrictions={restrictions}
+        >
+          <div
+            className={backgroundClasses.join(" ")}
+            style={style}
+            tabIndex="0"
+            onKeyDown={this.keyboardEvent}
+            onFocus={this.toggleFocus}
+            onBlur={this.toggleFocus}
+          >
+            {!background && (
+              <div className="u-align-text--center">
+                <i className="p-icon--plus" />
+                <br />
+                Add background
+              </div>
+            )}
+          </div>
+        </FileInput>
+        {background && (
+          <Fragment>
+            <span
+              className="p-market-banner__remove"
+              role="button"
+              onClick={this.handleRemoveImageClick}
+            >
+              <i className="p-icon--delete" />
+            </span>
+            <div className="p-market-banner__change">Edit background</div>
+          </Fragment>
+        )}
+      </Fragment>
+    );
+  }
+
+  renderHelp() {
+    return (
+      <Fragment>
+        <p className="p-form-help-text">
+          Adding a featured banner will increase your chances of being featured
+          on snapcraft.io and in GNOME software but does not immediately make
+          you eligible to be featured.
+        </p>
+        <AccordionHelp name="banner restrictions">
+          <p>
+            <small>
+              Accepted image formats include: <b>JPEG & PNG files</b>
+              <br />
+              Min resolution: <b>720 x 240 pixels</b>
+              <br />
+              Max resolution: <b>3840 x 1440 pixels</b>
+              <br />
+              Recommended (legacy) size: <b>1218 x 240 pixels</b>
+              <br />
+              Aspect ratio: <b>1:3</b>
```

Comment:
I guess this should be 3:1 (width:height)

Reply:
I blindly followed the spec, reeeeaaaaalllllyyyy good catch, thanks @bartaz